common/strmatcher/ac_automaton_matcher.go

```diff
@@ -225,7 +225,8 @@ func (ac *ACAutomaton) Match(s string) bool {
 	// 2. the match string is through a fail edge. NOT FULL MATCH
 	// 2.1 Through a fail edge, but there exists a valid node. SUBSTR
 	for i := len(s) - 1; i >= 0; i-- {
-		idx := char2Index[s[i]]
+		chr := s[i]
+		idx := char2Index[chr]
```

Comment:
Can we check the array len here?

Reply:
yes, we can check array len can ensure safe.